coresimd/ppvt/api/bitwise_reductions.rs

```diff
@@ -0,0 +1,83 @@
+//! Implements portable bitwise vector reductions.
+
+macro_rules! impl_bitwise_reductions {
+    ($id:ident, $elem_ty:ident) => {
+        impl $id {
+            /// Lane-wise bitwise `and` of the vector elements.
+            pub fn and(self) -> $elem_ty {
```

Comment:
So all these 3 are provided by llvm. They are necessary to implement the reductions of boolean vectors (`all`,`any`,`none`), but maybe we shouldn't provide them for the integer types (llvm supports that though).

Reply:
I am leaving these here for now since I can just map them directly to the llvm intrinsics. We might just decide to never stabilize these and expose them only for boolean vectors via `all`,`any`,`none`.